packages/utils/telemetry-utils/src/batchedTelemetryHelper.ts

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import type {
+	ITelemetryGenericEventExt,
+	ITelemetryLoggerExt,
+	ITelemetryPerformanceEventExt,
+} from "./telemetryTypes.js";
+
+/**
+ * Expected type of the custom data passed into the logger.
+ */
+type ICustomDataMap = Record<string, number>;
+
+/**
+ * Helper class that accumulates custom data and sends it to the logger every time a specified number (i.e., `threshold`) of logs is reached.
+ */
+export class BatchedTelemetryHelper {
```

Comment:
I would recommend against using "Helper" in the name and documentation. It's a very generic term, and can usually be replaced with something more concrete.

Reply:
(https://github.com/microsoft/FluidFramework/blob/109d5a951e13bfa12a31680a5653c7536a824de8/packages/utils/telemetry-utils/src/sampledTelemetryHelper.ts#L147

I was also hesitant to use "Helper", but also wanted to conform with the existing `SampledTelemetryHelper` which offers somewhat similar functionality.

Reply:
I would still argue for giving it a more semantic name. We should also rename `SampledTelemetryHelper` in the future in my opinion. Might make sense to call this something like `TelemetryEventBatcher`.